### PR TITLE
[ACS-5511] create library playwright tests

### PR DESCRIPTION
**Pull request by azakrzewski-hy**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

Test for create library on ACA is written in Protractor

**What is the new behaviour?**

Test for create library is written in Playwright

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

Method: "goThroughPagesLookingForRowWithName()" has "isVisible()" - this is not exact as "expect()" but does validation.
Do you think after "goThroughPagesLookingForRowWithName()" method, validation of the row should be done in expect() still ? e.g.:
    await myLibrariesPage.dataTable.goThroughPagesLookingForRowWithName(libraryName);
    await expect(myLibrariesPage.dataTable.getRowByName(libraryName)).toBeVisible();

Or maybe we should put "goThroughPagesLookingForRowWithName()" inside expect() ? e.g.:
expect(myLibrariesPage.dataTable.goThroughPagesLookingForRowWithName(libraryName));

### CODE REVIEW

File: e2e/playwright/actions/src/tests/create-library.spec.ts [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { expect } from '@playwright/test';
+import { ApiClientFactory, getUserState, test } from '@alfresco/playwright-shared';
+import { libraryErrors } from '../../../../../projects/aca-playwright-shared/src/utils/library-errors';
```
Comment (MichalKinas) [inline e2e/playwright/actions/src/tests/create-library.spec.ts:27]: If this code comes from `aca-playwright-share` it should be imported as part of `@alfresco/playwright-shared`

Comment (azakrzewski-hy) [reply e2e/playwright/actions/src/tests/create-library.spec.ts:27]: good point

File: e2e/playwright/actions/src/tests/create-library.spec.ts [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+  const libraryDialogTitle = 'Create Library';
+  const libraryNameLebel = 'Name *';
+  const libraryIdLebel = 'Library ID *';
+  const libraryDescriptionLebel = 'Description';
+  const publicVisibility = 'Public';
+  const moderatedVisibility = 'Moderated';
+  const privateVisibility = 'Private';
+  const deleteAction = 'Delete';
+
+  const commonLibraryName = `playwright-library-${(Math.random() + 1).toString(36).substring(6)}`;
```
Comment (DenysVuika) [inline e2e/playwright/actions/src/tests/create-library.spec.ts:44]: we already have random utility that is safe:

```ts
import { Utils } from '@alfresco/aca-testing-shared';
const username = `user-${Utils.random()}`;
```

File: projects/aca-playwright-shared/src/page-objects/components/aca-header.component.ts [hunk 3]
```diff
@@ -28,6 +28,7 @@
  export class AcaHeader extends BaseComponent {
    private static rootElement = 'adf-toolbar';
    public createButton = this.getChild('[id="app.toolbar.create"]');
+   public viewDetails = this.getChild('[title="View Details"]');
```
Comment (DenysVuika) [inline projects/aca-playwright-shared/src/page-objects/components/aca-header.component.ts:31]: this button should already have an id, like the "createButton"

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: e2e/playwright/actions/src/tests/create-library.spec.ts (hunk 2 above)
Comment (azakrzewski-hy) [reply e2e/playwright/actions/src/tests/create-library.spec.ts:44]: doesn't act-testing-shared is a protractor part? You mean, that we better create similar for playwright use ?

File: e2e/playwright/actions/src/tests/create-library.spec.ts [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+  test.afterAll(async () => {
+    await apiClientFactory.sites.deleteSite(commonLibraryId, { permanent: true });
+  });
+
+  test('[C280024] Create Library dialog UI', async ({ myLibrariesPage }) => {
+    await myLibrariesPage.acaHeader.createButton.click();
+    await myLibrariesPage.matMenu.createLibrary.click();
+
+    await expect(myLibrariesPage.libraryDialog.getDialogTitle(libraryDialogTitle)).toBeVisible();
```
[the remark this answers is not in the document]
Comment (azakrzewski-hy) [reply e2e/playwright/actions/src/tests/create-library.spec.ts:75]: good point

File: projects/aca-playwright-shared/src/page-objects/components/aca-header.component.ts (hunk 3 above)
Comment (azakrzewski-hy) [reply projects/aca-playwright-shared/src/page-objects/components/aca-header.component.ts:31]: can't find any 
Screenshot 2023-07-11 at 07 42 53

File: projects/aca-playwright-shared/src/page-objects/components/adf-info-drawer.component.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import { Page } from '@playwright/test';
+
+export class AdfInfoDrawerComponent extends BaseComponent {
+  private static rootElement = 'adf-info-drawer';
+
+  constructor(page: Page) {
+    super(page, AdfInfoDrawerComponent.rootElement);
+  }
+
+  public getFieldData = (labelText: string) => this.getChild('.mat-form-field-wrapper', { hasText: labelText });
```
[the remark this answers is not in the document]
Comment (azakrzewski-hy) [reply projects/aca-playwright-shared/src/page-objects/components/adf-info-drawer.component.ts:35]: here I need to use your advice as we can see separate class names for a label and an element but validating them in separate calls do not prove that one is connected with the other. Looking at the screen what would be the best locator to use, to make sure that such label gets such element input ?
Screenshot 2023-07-11 at 08 02 17

File: projects/aca-playwright-shared/src/page-objects/components/dialogs/adf-library-dialog.component.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+   * @param libraryIdInput input for Library ID field
+   * @param descriptionInput input for Description field
+   * @param visibility visibility of the library
+   */
+    async createLibraryWithNameAndId(nameInput: string, libraryIdInput: string, descriptionInput?: string, visibility?: string): Promise<void> {
+      await this.getLabelText('Name *').fill(nameInput);
+      await this.getLabelText('Library ID *').clear();
+      await this.getLabelText('Library ID *').fill(libraryIdInput);
+      if (descriptionInput) {
+      await this.getLabelText('Description').fill(descriptionInput);
```
[the remark this answers is not in the document]
Comment (azakrzewski-hy) [reply projects/aca-playwright-shared/src/page-objects/components/dialogs/adf-library-dialog.component.ts:55]: not sure if this is only my Visual Studio Code, but its so smart that all the time is screaming about indentations. VS provided this structure when I've agreed to "Quick fix..." option. Sometimes is to quick indeed ;)

File: projects/aca-playwright-shared/src/page-objects/components/adf-info-drawer.component.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (azakrzewski-hy) [reply projects/aca-playwright-shared/src/page-objects/components/adf-info-drawer.component.ts]: data-automation-id added in [d379c1f](https://github.com/Alfresco/alfresco-content-app/pull/3318/commits/d379c1f0d84fc42c342d94046025b0d7efbc672e)

File: projects/aca-playwright-shared/src/page-objects/components/aca-header.component.ts (hunk 3 above)
Comment (azakrzewski-hy) [reply projects/aca-playwright-shared/src/page-objects/components/aca-header.component.ts:31]: this is not an ACA implementation but ADF is blocked by CF at the moment

File: projects/aca-playwright-shared/src/page-objects/components/dialogs/adf-library-dialog.component.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (azakrzewski-hy) [reply projects/aca-playwright-shared/src/page-objects/components/dialogs/adf-library-dialog.component.ts]: done in [4fe33db](https://github.com/Alfresco/alfresco-content-app/pull/3318/commits/4fe33dbd179934f852b31b90178d0a3a5c6bb24a)

File: e2e/playwright/actions/src/tests/create-library.spec.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (azakrzewski-hy) [reply e2e/playwright/actions/src/tests/create-library.spec.ts]: done in [4fe33db](https://github.com/Alfresco/alfresco-content-app/pull/3318/commits/4fe33dbd179934f852b31b90178d0a3a5c6bb24a)

File: e2e/playwright/actions/src/tests/create-library.spec.ts [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { expect } from '@playwright/test';
+import { libraryErrors, ApiClientFactory, getUserState, test } from '@alfresco/playwright-shared';
+import { SiteBodyCreate } from '@alfresco/js-api';
+import { GeneralUtils } from '../../../../../projects/aca-playwright-shared/src/utils/general-utils';
```
Comment (DenysVuika) [inline e2e/playwright/actions/src/tests/create-library.spec.ts:28]: incorrect import

Comment (azakrzewski-hy) [reply e2e/playwright/actions/src/tests/create-library.spec.ts]: import updated in [36188b9](https://github.com/Alfresco/alfresco-content-app/pull/3318/commits/36188b94ed6bd7e6d6c249332061c343142090cd)

File: projects/aca-playwright-shared/src/utils/general-utils.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+export class GeneralUtils {
+
+
+  static random(): string {
+    return (Math.random() + 1).toString(36).substring(6);
```
Comment (DenysVuika) [inline projects/aca-playwright-shared/src/utils/general-utils.ts:30]: there's already a random util in another PR with different implementation?

Comment (DenysVuika) [reply projects/aca-playwright-shared/src/utils/general-utils.ts:30]: https://github.com/Alfresco/alfresco-content-app/pull/3335/files#diff-08391580e8082276eed287862446386e71d15e46a0778704448447b318f6019f

Comment (azakrzewski-hy) [reply projects/aca-playwright-shared/src/utils/general-utils.ts:30]: @DenysVuika  it looks like mentioned PR is younger than this one so should I update to it anyway ?

**Review by DenysVuika — APPROVED** [review]

File: e2e/playwright/actions/src/tests/create-library.spec.ts [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+import { SiteBodyCreate } from '@alfresco/js-api';
+
+test.use({ storageState: getUserState('hruser') });
+test.describe('Create Libraries ', () => {
+  const apiClientFactory = new ApiClientFactory();
+  let randomLibraryName: string;
+  let randomLibraryId: string;
+  let randomLibraryDescription: string;
+  const libraryDialogTitle = 'Create Library';
+  const libraryNameLebel = 'Name *';
```
Comment (MichalKinas) [inline e2e/playwright/actions/src/tests/create-library.spec.ts:36]: I think there is a typo, should be `libraryNameLabel`

File: e2e/playwright/actions/src/tests/create-library.spec.ts [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+  const libraryIdLebel = 'Library ID *';
```
Comment (MichalKinas) [inline e2e/playwright/actions/src/tests/create-library.spec.ts:37]: Same here

File: e2e/playwright/actions/src/tests/create-library.spec.ts [hunk 11]
```diff
@@ -0,0 +1,263 @@
...
+  const libraryDescriptionLebel = 'Description';
```
Comment (MichalKinas) [inline e2e/playwright/actions/src/tests/create-library.spec.ts:38]: And here

**Review by MichalKinas — APPROVED** [review]

File: e2e/playwright/actions/src/tests/create-library.spec.ts (hunk 9 above)
Comment (azakrzewski-hy) [reply e2e/playwright/actions/src/tests/create-library.spec.ts]: change provided in [a9a9a98](https://github.com/Alfresco/alfresco-content-app/pull/3318/commits/a9a9a98483d2b0cba804fbecd1a3f2d79a8345a6)

File: e2e/playwright/actions/src/tests/create-library.spec.ts (hunk 10 above)
Comment (azakrzewski-hy) [reply e2e/playwright/actions/src/tests/create-library.spec.ts]: change provided in [a9a9a98](https://github.com/Alfresco/alfresco-content-app/pull/3318/commits/a9a9a98483d2b0cba804fbecd1a3f2d79a8345a6)

File: e2e/playwright/actions/src/tests/create-library.spec.ts (hunk 11 above)
Comment (azakrzewski-hy) [reply e2e/playwright/actions/src/tests/create-library.spec.ts]: change provided in [a9a9a98](https://github.com/Alfresco/alfresco-content-app/pull/3318/commits/a9a9a98483d2b0cba804fbecd1a3f2d79a8345a6)

**Pull request merged**